src/Effects.js

```diff
@@ -1445,6 +1449,67 @@ module.exports = class Effects {
         }
       }
     };
+
+    this.emojis = {
+      emojiList : [],
+      emojiTypes : [],
+
+      init : function () {
+        this.imageLovestruck = p5.createGraphics(100, 100);
+        this.imageLovestruck.scale(3);
+        drawLovestruck(this.imageLovestruck.drawingContext);
+        this.emojiTypes.push(this.imageLovestruck);
+
+        this.imageSmiley = p5.createGraphics(100, 100);
+        this.imageSmiley.scale(3);
+        drawSmiley(this.imageSmiley.drawingContext);
+        this.emojiTypes.push(this.imageSmiley);
+
+        this.imageStarstruck = p5.createGraphics(100, 100);
+        this.imageStarstruck.scale(3);
+        drawStarstruck(this.imageStarstruck.drawingContext);
+        this.emojiTypes.push(this.imageStarstruck);
+
+        this.imageTickled = p5.createGraphics(100, 100);
+        this.imageTickled.scale(3);
+        drawTickled(this.imageTickled.drawingContext);
+        this.emojiTypes.push(this.imageTickled);
+
+        this.imageWink = p5.createGraphics(100, 100);
+        this.imageWink.scale(3);
+        drawWink(this.imageWink.drawingContext);
+        this.emojiTypes.push(this.imageWink);
+      },
+
+      draw : function () {
+        if (p5.frameCount % 10 === 0) {
+          this.emojiList.push(new this.emoji(this.emojiTypes, this.emojiList));
+        }
+        for (let aEmoji of this.emojiList) {
+          aEmoji.update();
+          aEmoji.display();
+        }
+      },
+
+      emoji : function (emojiTypes, emojiList) {
```

Comment:
To simplify the code and make it more readable, I don't think we need the 'emoji' function. If we store the `x`, `y`, `size`, and `image` in an object, similar to how we initialize smiles here: https://github.com/code-dot-org/dance-party/blob/083579e8dec56373028c8b935e089fd5b40c32d5/src/Effects.js#L1340, and combine the update and display code into the `draw` function, similar to how we draw smiles here: https://github.com/code-dot-org/dance-party/blob/083579e8dec56373028c8b935e089fd5b40c32d5/src/Effects.js#L1355, we can get rid of the emoji function. This will make this effect more similar to the other foreground effects and improve the readability.

Reply:
Good idea! I'll do that.